ibis-server/app/mdl/rewriter.py

```diff
@@ -95,14 +95,14 @@ def __init__(self, manifest_str: str, function_path: str):
         self.manifest_str = manifest_str
         self.function_path = function_path
 
-    def rewrite(self, sql: str) -> str:
+    async def rewrite(self, sql: str) -> str:
         try:
             extractor = get_manifest_extractor(self.manifest_str)
             tables = extractor.resolve_used_table_names(sql)
             manifest = extractor.extract_by(tables)
             manifest_str = to_json_base64(manifest)
             session_context = get_session_context(manifest_str, self.function_path)
-            return session_context.transform_sql(sql)
+            return await to_thread.run_sync(session_context.transform_sql, sql)
```

Comment:
It's potentially a nice improvement. `transform_sql` is an async Rust method, but we wrap it into a sync method. I found pyo3 has [async interface](https://pyo3.rs/v0.13.2/ecosystem/async-await). Maybe we can change to it in the future.